### DISCUSSION TITLE
White page after upgrade

### DISCUSSION THREAD

**Discussion by CountMurphy** [firefly-iii/firefly-iii#8722] [General]
After updating to 6.1.12, I get a white page. Looking at the server logs I'm seeing:

```
PHP Fatal error:  Uncaught ReflectionException: Class "hash" does not exist     in /usr/share/webapps/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php
```

This is a self managed instance (using the aur package in Arch linux). Downgrading to 6.1.10 works again.

**Comment by JC5** [comment]
I'm not sure, how is the aur package in Arch linux made?

**Reply by CountMurphy** [reply]
Here is the install script for it. Basically it copies the files to a "package" directory ($pkgdir) and sets the right permissions. Once the package is complete, the package manager will install everything in the $pkgdir directory to disk (/usr/share/webapps/firefly-ii mostly). I can try to answer any questions about the code below.

```
package() {
    cd "$srcdir/$pkgname-$pkgver"
    COMPOSER_ALLOW_SUPERUSER=1 composer install --no-scripts --no-dev --ignore-platform-reqs

    install -d "$pkgdir/usr/share/webapps/$pkgname" "$pkgdir/usr/share/licenses/$pkgname" "$pkgdir/etc/webapps/$pkgname"
    cp -rv * "$pkgdir/usr/share/webapps/$pkgname"
    install -D "$srcdir/$pkgname-$pkgver/LICENSE" "$pkgdir/usr/share/licenses/$pkgname"

    cp -v .env.example "$pkgdir/etc/webapps/$pkgname/config.env"

    ln -s "/etc/webapps/$pkgname/config.env" "$pkgdir/usr/share/webapps/$pkgname/.env"
    rm -rf "$pkgdir/usr/share/webapps/$pkgname/bootstrap/cache"

    mkdir -p "$pkgdir/var/cache/$pkgname"
    chown http:http "$pkgdir/var/cache/$pkgname"
    ln -s "/var/cache/$pkgname" "$pkgdir/usr/share/webapps/$pkgname/bootstrap/cache"

    mkdir -p "$pkgdir/var/lib"
    mv "$pkgdir/usr/share/webapps/$pkgname/storage" "$pkgdir/var/lib/firefly-iii"
    chown -R http:http "$pkgdir/var/lib/firefly-iii"
    chmod 775 "$pkgdir/var/lib/firefly-iii"
    ln -s "/var/lib/firefly-iii" "$pkgdir/usr/share/webapps/$pkgname/storage"
}
```

At this point I don't know if I'm missing some sort of dependency, or the install script is broken. 

If the snippet above isn't enough, I can post the entire script.

**Reply by JC5** [reply]
If you're downloading the raw source, that could be the issue. Check out the release notes of the latest release. The source no longer includes the generated JS files. That could be it?

**Reply by CountMurphy** [reply]
I tried replacing the tar.gz file with the zip file instead earlier today. Same result. If it was JS I'd expect the error to show up in my browser console. The error above was from an apache server error log. That said, trying again to make sure I didn't do something dumb.

EDIT:
Yup, even specifying the correct zip results in the same error about hash missing.

**Reply by CountMurphy** [reply]
I fixed the issue if anyone else ever runs into it. In your installation directory, run `composer update`

**Reply by JC5** [reply]
Thanks for mentioning it! The new installation instructions should prevent this for new users, but if you run into this problem, composer install or composer update should do the trick.
